File: AbilityChartView/app/src/main/java/tactical/twerk/abilitychartview/MainActivity.java
package tactical.twerk.abilitychartview;


import android.os.Bundle;
import android.text.Editable;
import android.text.TextUtils;
import android.text.TextWatcher;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import androidx.appcompat.app.AppCompatActivity;
import java.util.Random;

public class MainActivity extends AppCompatActivity {

    private AbilityChartView mPpv;
    private Button mBtnSwitch;
    private Button mBtnLess,mBtnMore;
    private EditText mEtSides;
    private int mSides = -1;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mPpv = findViewById(R.id.ppv);
        mBtnSwitch =  findViewById(R.id.btnSwitch);
        mBtnLess = findViewById(R.id.btn_less);
        mBtnMore = findViewById(R.id.btn_more);
        mEtSides = findViewById(R.id.et_number);
        mBtnLess.setOnClickListener(new View.OnClickListener() {
            @Override public void onClick(View view) {
                if (mSides==-1 || mSides<=3){
                    return;
                }
                mSides -=1;
                mEtSides.setText(String.valueOf(mSides));
            }
        });
        mBtnMore.setOnClickListener(new View.OnClickListener() {
            @Override public void onClick(View view) {
                if (mSides==-1){
                    mSides = 3;
                }else{
                    mSides +=1;
                }
                mEtSides.setText(String.valueOf(mSides));
            }
        });
        mEtSides.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {

            }

            @Override public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {

            }

            @Override public void afterTextChanged(Editable editable) {
                if (null==editable || TextUtils.isEmpty(editable.toString()) ||Integer.valueOf(editable.toString())<3){
                    mSides=-1;
                }else {
                    mSides = Integer.valueOf(editable.toString());
                }
            }
        });

        mBtnSwitch.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                Random random = new Random();
                int sides = mSides >=3 ?mSides:random.nextInt(5)+10;
                float[] mProgressValues = new float[sides];
                for (int i = 0; i < sides; i++) {
                    mProgressValues[i] = random.nextFloat();
                    if (random.nextInt() % 3 == 0) {
                        mProgressValues[i] = 1;
                    }
                }
                String []mLabels = new String[sides];
                for (int i = 0; i < sides; i++) {
                    mLabels[i] = "label" + i;
                }

                mPpv.initial(sides,mProgressValues,mLabels);
                mPpv.animateProgress();
            }
        });

    }


}

File: app/src/main/java/io/github/tacticaltwerking/demo/MainActivity.kt
package io.github.tacticaltwerking.demo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.text.TextUtils
import androidx.core.widget.addTextChangedListener
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*

class MainActivity : AppCompatActivity() {


    var mSides = -1

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btn_less.setOnClickListener {
            if (mSides >=3) {
                mSides -= 1
                et_sides.setText(mSides.toString())
            }
        }

        btn_more.setOnClickListener {
            if (mSides == -1) {
                mSides = 3
            } else {
                mSides += 1
            }
            et_sides.setText(mSides.toString())
        }

        et_sides.addTextChangedListener {
            mSides = if (null == it
                || TextUtils.isEmpty(it.toString())
                || Integer.valueOf(it.toString()) < 3
            ) {
                -1
            } else {
                Integer.valueOf(it.toString())
            }
        }

        btnSwitch.setOnClickListener {
            val random = Random()
            val sides = if (mSides >= 3) mSides else random.nextInt(5) + 10
            val mProgressValues = FloatArray(sides)
            for (i in 0 until sides) {
                mProgressValues[i] = random.nextFloat()
                if (random.nextInt() % 3 == 0) {
                    mProgressValues[i] = 1f
                }
            }
            val mLabels = arrayOfNulls<String>(sides)
            for (i in 0 until sides) {
                mLabels[i] = "label$i"
            }

            ppv.initial(sides, mProgressValues, mLabels)
            ppv.animateProgress()
        }
    }
}
